web/client/components/map/leaflet/ScaleBar.jsx

```diff
@@ -41,6 +43,14 @@ class ScaleBar extends React.Component {
                     mainControl.parentNode.removeChild(mainControl);
                 }
             }
+            // as there is no direct way to set a custom class to scalebar in leaflet -->
+            // use DOM munipulating to get scalebar element and set the needed custom class that holds your style
+            // here the needed for mobile mode in case of widgetsTray widget is existing on map
+            const widgetsTrayBtnExistInMobileMode = this.props.isMobile && document.querySelector('.widgets-tray');
+            if (widgetsTrayBtnExistInMobileMode) {
+                const scaleBarElement  = document.querySelector('.leaflet-control-scale');
+                scaleBarElement .classList.add('mobile-map-with-widgets-tray');
+            }
```

Comment:
We should avoid dom manipulation like this one, in particular we cannot bind an hardcoded class of a different component inside another one.

My proposal is to remove this part and related class and move the logic to the [WidgetTray](https://github.com/geosolutions-it/MapStore2/blob/master/web/client/plugins/widgets/WidgetsTray.jsx) plugin bacause it is already connected to the state and we could move the margin bottom based on some condition:

- if it is mobile device and not 3D visualization apply a margin bottom of 60
- other cases use the default value, currently 32

Expected results (cc @tdipisa )

mobile 2D
![image](https://github.com/geosolutions-it/MapStore2/assets/19175505/504fa144-329b-40e9-827b-7a740609eee3)

mobile 2D with widgets
![image](https://github.com/geosolutions-it/MapStore2/assets/19175505/41ec75c5-2c84-44fd-a8c2-fe7e49033520)

mobile 3D with widgets
![image](https://github.com/geosolutions-it/MapStore2/assets/19175505/fecdbd6a-8679-40b8-82e5-896535910254)

Reply:
ok that's also fine @allyoucanmap